src/motion_detector.py

```diff
@@ -78,10 +105,20 @@ def reconfigure(
         self.sensitivity = config.attributes.fields["sensitivity"].number_value
         if self.sensitivity == 0:
             self.sensitivity = 0.9
-        self.min_box_size = config.attributes.fields["min_box_size"].number_value
+
+        # Store all possible box size constraints
+        self.min_box_size = config.attributes.fields.get("min_box_size")
+        if self.min_box_size is not None:
+            self.min_box_size = self.min_box_size.number_value
```

Comment:
I wonder if this pattern could be simplified. It almost seems like `fields` returns a valid, 0-like value even if you don't specify anything in the config, so I don't need to worry about raising a `KeyError` if I look up a key that doesn't exist. Then again, especially for maximum sizes, I don't want to accidentally get a 0 if there's nonexistent data, and I want to accurately keep the 0 if that's what the user actually specified (perhaps as a way to disable all detections). Maybe this is okay after all...

Reply:
Yes, this has been simplified! Thanks to Khari pointing out that `config.attributes.fields` is indeed _not_ a dict!